### ISSUE TITLE
[daikin] Bug / Missing value Fanrate f_rate and configurable value for AUTO mode

### ISSUE THREAD

**Issue by LuiSauberhorn** [openhab/openhab-addons#15375]
Hi,

I am currently setting up two Daikin AirCons. When testing different setup and configs, I found two cases that cause problems. One seems to be rather a bug, the other one is a feature request.

Setup
I have two things configured:

```
/* stylish wall mount device*/
daikin:ac_unit:sleeping_room_ac "Klimaanlage Schlafzimmer" [
  host="192.x.y.9"
]

/* BRP069A81 */
daikin:ac_unit:living_room_ac "Klimaanlage Wohnzimmer" [
  host="192.y.x.8"
]
```
**BUG: DEHUMIDIFIER and Fan Rate**
When setting the mode to DEHUMIDIFIER (either directly via http-API or the binding), the daikin sets stemp=-- and f_rate=-. While stemp=-- does not cause any issues and is interpreted correctly, the fact that is set to f_rate=- causes a debug message:

```
2023-06-24 23:10:27.937 [DEBUG] [g.daikin.internal.api.Enums$FanSpeed] - Unexpected FanSpeed value of “-”
```
The API reply for http://192.x.y.8/aircon/get_control_info in DEHUMIDIFIER Mode is
`ret=OK,pow=0,mode=2,stemp=–,shum=–,dt1=25.0,dt2=–,dt3=25.0,dt4=25.0,alert=0,adv=,f_rate=-,dfr1=A,dfr2=-,dfr3=A,dfr4=A,dfr6=3`
It would be great if the “-” could be considered as valid value for f_rate.

**Feature: Correct Mode Value for**
As written above, I am using a BRP069A81 adapter to connect the AirCon. I know that this is formally not supported but it works almost well.
While everything works fine, there is a slight difference in how the mode value is interpreted. While my other Daikin uses the value 0 for AUTO mode (which works fine with the binding), the BRP069A81 uses the value 1 instead. This is a value the binding does not understand and as 0 is set instead of the required 1, the Aircon cannot be controlled when it is in AUTO mode.
Here, it would be great to have an optional configuration option in the daikin:ac_unit:* thing to use the value 1 instead of 0 for mode = AUTO.

Happy to provide more details & discuss.

Thanks, Lui

**Comment by lsiepel** [comment]
Regarding the fanspeed, the - is not listed in the corrospoding `Fanspeed` enum. What should the - represent? 
https://github.com/openhab/openhab-addons/blob/main/bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/api/Enums.java#L61-L67

Value 1 is missing from the `Mode` enum, so well, yeah it is unsupported :-)
https://github.com/openhab/openhab-addons/blob/main/bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/api/Enums.java#L61-L67
But when received, it converts all unknown values (also 1) to 0/AUTO. I don't see a problem until this point, as the UI/channels etc match the functional state. But now when thinking about it, when sending a command, it also writes a mode=0 to the API, and the device probably ignores this as it is not the expected 1.

There are several solutions, but maybe @caffineehacker as codeowner can comment?

**Comment by jimtng** [comment]
> ```
> 2023-06-24 23:10:27.937 [DEBUG] [g.daikin.internal.api.Enums$FanSpeed] - Unexpected FanSpeed value of “-”
> ```
> 
> 
>     
>   
> 
> The API reply for http://192.x.y.8/aircon/get_control_info in DEHUMIDIFIER Mode is
> `ret=OK,pow=0,mode=2,stemp=–,shum=–,dt1=25.0,dt2=–,dt3=25.0,dt4=25.0,alert=0,adv=,f_rate=-,dfr1=A,dfr2=-,dfr3=A,dfr4=A,dfr6=3`
> It would be great if the “-” could be considered as valid value for f_rate.

For this, you could simply set the log level to INFO or DEFAULT so you don't see the DEBUG messages. 



> **Feature: Correct Mode Value for**
> As written above, I am using a BRP069A81 adapter to connect the AirCon. I know that this is formally not supported but it works almost well.
> While everything works fine, there is a slight difference in how the mode value is interpreted. While my other Daikin uses the value 0 for AUTO mode (which works fine with the binding), the BRP069A81 uses the value 1 instead. This is a value the binding does not understand and as 0 is set instead of the required 1, the Aircon cannot be controlled when it is in AUTO mode.
> Here, it would be great to have an optional configuration option in the daikin:ac_unit:* thing to use the value 1 instead of 0 for mode = AUTO.

@LuiSauberhorn could you please test this jar to see if it solved the problem of AUTO mode using the value of 1 for you:

https://github.com/jimtng/openhab-addons/releases/download/daikin-automode-0.1/org.openhab.binding.daikin-4.1.0-SNAPSHOT.jar

This bundle version should be `4.1.0.202308171145`

**Comment by jimtng** [comment]
> Regarding the fanspeed, the - is not listed in the corrospoding `Fanspeed` enum. What should the - represent?

I would imagine that `-` represents: `not applicable`. In other words, under that mode, the fan speed cannot be controlled. Just my guess.

**Comment by LuiSauberhorn** [comment]
> @LuiSauberhorn could you please test this jar to see if it solved the problem of AUTO mode using the value of 1 for you:
> 
> 
> EDIT: Updated to v0.2 below:
> 
> 
> https://github.com/jimtng/openhab-addons/releases/download/daikin-automode-0.2/org.openhab.binding.daikin-4.1.0-SNAPSHOT.jar

Sorry, I was out on holidays. I am pack and will test tonight. Feedback follows...

**Comment by LuiSauberhorn** [comment]
Hi @jimtng,

everything tested, looks good. At first start after installation, I had the feeling that it was not working instantly (here was a message in the logs like "Cannot set AUTO mode with '0' or '1'", but I did not keep the log. After seeing the value manually and one open hab restart, it now works perfectly. may have been a caching issue, I don't know.

I can also confirm that it works "in both directions". I have one daikin AC that requires 0 as auto value and another one that requires 1. Both work in mixed mode.  

Thanks for taking care!!!

I an also quite sure that '-' means n/a, as it only appears for modes where a temperature does not make any sense.

**Comment by jimtng** [comment]
Reopening this to track the corresponding PR.

**Comment by jimtng** [comment]
@LuiSauberhorn thanks for the feedback. I've made some minor tweaks in the code, removed that warning, but added a DEBUG message stating which mode value the binding will use. 

Would you please test the revised version (v0.3) below
Binding version will be `4.1.0.202308182301`

https://github.com/jimtng/openhab-addons/releases/download/daikin-automode-0.3/org.openhab.binding.daikin-4.1.0-SNAPSHOT.jar

**Comment by LuiSauberhorn** [comment]
Hi @jimtng 

first Observations:
1) I get a lot of DEBUG messages like
`2023-08-19 09:11:52.308 [DEBUG] [ing.daikin.internal.DaikinWebTargets] - HTTP request error on attempt #1: http://192.168.191.128/aircon/get_control_info java.net.NoRouteToHostException: Keine Route zum Zielrechner`
No route to host is OK, as the two air-conditions are powered on demand. There is no route. So all good here.
2) Initial start is still a little bumpy. As you see below, I saw a mode value of 7, which could not be parsed:
```
2023-08-19 09:31:30.537 [TRACE] [ing.daikin.internal.DaikinWebTargets] - Calling url: http://192.168.191.129/aircon/get_control_info
2023-08-19 09:31:30.644 [TRACE] [ding.daikin.internal.api.ControlInfo] - Parsing string: "ret=OK,pow=1,mode=7,adv=,stemp=24.5,shum=0,dt1=24.5,dt2=M,dt3=25.0,dt4=25.0,dt5=25.0,dt7=24.5,dh1=0,dh2=50,dh3=0,dh4=0,dh5=0,dh7=0,dhh=50,b_mode=7,b_stemp=24.5,b_shum=0,alert=255,f_rate=A,f_dir=0,b_f_rate=A,b_f_dir=0,dfr1=A,dfr2=5,dfr3=5,dfr4=5,dfr5=5,dfr6=5,dfr7=A,dfrh=5,dfd1=0,dfd2=0,dfd3=0,dfd4=0,dfd5=0,dfd6=0,dfd7=0,dfdh=0,dmnd_run=0,en_demand=0"
2023-08-19 09:31:30.645 [DEBUG] [nding.daikin.internal.api.Enums$Mode] - Unexpected Mode value of "7"
The `.129 device` is the one that worked with `AUTO mode 0`. However, according to documentation 7 also represents auto mode, see [here](https://github.com/ael-code/daikin-control).
```
2023-08-19 09:33:12.771 [TRACE] [ing.daikin.internal.DaikinWebTargets] - Calling url: http://192.168.191.128/aircon/set_control_info?mode=1&f_dir_ud=0&f_rate=A&f_dir_lr=0&shum=&pow=1&stemp=24.0
2023-08-19 09:33:12.861 [TRACE] [ing.daikin.internal.DaikinWebTargets] - Calling url: http://192.168.191.129/aircon/set_control_info?mode=1&f_rate=A&shum=0&pow=1&stemp=24.5&f_dir=0

3) After a while, all converges.
for the `.128 device` using AUTO mode 1:
2023-08-19 09:41:15.156 [TRACE] [ing.daikin.internal.DaikinWebTargets] - Calling url: http://192.168.191.128/aircon/get_control_info
2023-08-19 09:41:15.194 [TRACE] [ding.daikin.internal.api.ControlInfo] - Parsing string: "ret=OK,pow=1,mode=1,stemp=24.0,shum=--,dt1=24.0,dt2=--,dt3=24.0,dt4=24.0,alert=0,adv=,f_rate=A,dfr1=A,dfr2=-,dfr3=A,dfr4=3,dfr6=3"

for the `.129 device` using AUTO mode 0/1/7:
2023-08-19 09:42:55.610 [TRACE] [ing.daikin.internal.DaikinWebTargets] - Calling url: http://192.168.191.128/aircon/get_control_info
2023-08-19 09:42:55.618 [TRACE] [ding.daikin.internal.api.ControlInfo] - Parsing string: "ret=OK,pow=1,mode=1,stemp=24.0,shum=--,dt1=24.0,dt2=--,dt3=24.0,dt4=24.0,alert=0,adv=,f_rate=A,dfr1=A,dfr2=-,dfr3=A,dfr4=3,dfr6=3"
Interestingly, we .129 now also reports AUTO mode = 1. I can provide more details tonight.

**Comment by jimtng** [comment]
Is there any difference between mode=1 and mode=7?
As I understand it from your reports, units with mode=1 cannot be set to AUTO mode by sending mode=0, correct?

**Comment by jimtng** [comment]
@LuiSauberhorn Please test v0.4 (Version `4.1.0.202308191032`)
https://github.com/jimtng/openhab-addons/releases/download/daikin-automode-0.4/org.openhab.binding.daikin-4.1.0-SNAPSHOT.jar

This version should also accept mode=7 as valid and send back mode=7 when setting the mode to AUTO.

The only caveat is if the aircon is not currently in Auto mode when the binding first starts, then you set it to AUTO through the binding, it will not know that the aircon uses mode=7 and will send mode=0. From what you've said, I believe this would be fine?

**Comment by LuiSauberhorn** [comment]
Not quite sure. The modern one (`.129` in the logs) from 2023 should for sure accept 0 and sometimes sends a 7. the critical one is `.128` from 2020, that accept only 1.
I will try all combinations manually tonight, i.e. sending a start with AUTO value 0, 1 and 7 to `.128` and `.129`. Let's see how they behave.
First thought, to be tested: Starting with AUTO = 0, in case that creates an error, resend the message with AUTO = 1 could be an option. That should cover both device types. Alternatively, a thing for the BRP069A81 connector could be an option...

I come back to you tonight.

**Comment by jimtng** [comment]
> First thought, to be tested: Starting with AUTO = 0, in case that creates an error, resend the message with AUTO = 1 could be an option. That should cover both device types.

This is exactly how it's done in the PR. Furthermore, once it has figured out what the adapter will accept, it remembers it so it won't need to try mode=0 first every time, and immediately just send mode=1 or mode=7 in the future.

**Comment by jimtng** [comment]
@LuiSauberhorn If you manually called the set_control_info with an invalid mode= (but with valid parameters for everything else), does it return anything other than `ret=OK`? If it does, I won't have to call get_control_info.

**Comment by LuiSauberhorn** [comment]
OK, here we are. The short answer is: MODE = 1 works for both devices as AUTO.
No clue what the difference between 0,1 and 7 is, the modes seem to behave similar.
In case I send a value that does not work, I always get a `ret=PARAM NG` back.

The long answer is:
```
http://192.168.191.128/aircon/get_control_info
ret=OK,pow=0,mode=1,stemp=24.0,shum=--,dt1=24.0,dt2=--,dt3=24.0,dt4=24.0,alert=0,adv=,f_rate=A,dfr1=A,dfr2=-,dfr3=3,dfr4=3,dfr6=3


http://192.168.191.128/aircon/set_control_info?mode=0&f_dir_ud=0&f_rate=A&f_dir_lr=0&shum=&pow=0&stemp=24.0
ret=PARAM NG

http://192.168.191.128/aircon/set_control_info?mode=1&f_dir_ud=0&f_rate=A&f_dir_lr=0&shum=&pow=0&stemp=24.0
ret=OK,adv=

http://192.168.191.128/aircon/set_control_info?mode=7&f_dir_ud=0&f_rate=A&f_dir_lr=0&shum=&pow=0&stemp=24.0
ret=PARAM NG




http://192.168.191.129/aircon/get_control_info
ret=OK,pow=0,mode=1,adv=,stemp=24.5,shum=0,dt1=24.5,dt2=M,dt3=25.0,dt4=25.0,dt5=25.0,dt7=24.5,dh1=0,dh2=50,dh3=0,dh4=0,dh5=0,dh7=0,dhh=50,b_mode=1,b_stemp=24.5,b_shum=0,alert=255,f_rate=A,f_dir=0,b_f_rate=A,b_f_dir=0,dfr1=A,dfr2=5,dfr3=5,dfr4=5,dfr5=5,dfr6=5,dfr7=A,dfrh=5,dfd1=0,dfd2=0,dfd3=0,dfd4=0,dfd5=0,dfd6=0,dfd7=0,dfdh=0,dmnd_run=0,en_demand=0

http://192.168.191.129/aircon/set_control_info?mode=0&f_dir_ud=0&f_rate=A&f_dir_lr=0&shum=&pow=0&stemp=24.0
ret=OK,adv=

http://192.168.191.129/aircon/set_control_info?mode=1&f_dir_ud=0&f_rate=A&f_dir_lr=0&shum=&pow=0&stemp=24.0
ret=OK,adv=

http://192.168.191.129/aircon/set_control_info?mode=7&f_dir_ud=0&f_rate=A&f_dir_lr=0&shum=&pow=0&stemp=24.0
ret=OK,adv=
```

Does that help?

**Comment by jimtng** [comment]
Thanks, that helps. It means I don't need to issue a get_control_info after trying to set the mode to see whether it's accepted. My controller (Airbase - so it's not relevant here) will always return ret=OK regardless of what mode=XX I gave it.

Please test v0.5 bundle version `4.1.0.202308191340`.
https://github.com/jimtng/openhab-addons/releases/download/daikin-automode-0.5/org.openhab.binding.daikin-4.1.0-SNAPSHOT.jar

**Comment by LuiSauberhorn** [comment]
So, tested it...

In general it looks good, AUTO mode detection happens according to the value that is send. However... - after a couple of hours testing, my final conclusion is: we got the logic wrong. What do I observe:

- The old unit `.128` accepts only mode=1 for AUTO and only sends that value. Obviously, the API has not heard of parameters 0 and 7.
- The new unit `.129` accepts modes 0, 1, 7 for AUTO mode and sends either 0 or 7.

Now, I observe the following:
- When I send 0 after boot of the aircon, it starts with cooling (until exceeding the thresholds), when I send 7 after boot it starts with heating (until exceeding the thresholds).
- When I send 1, it it checks the actual temperature of the room and does what applies according to the temperature.
- When it starts, it always sends 7 first.

**So, my understanding is the following: Sending 1 as mode for AUTO parameter is always the right behavior.**

This is supported by both device types. The old one can only tell us "I am in AUTO mode" by sending mode=1. The new one is smarter. It tells us "I am in AUTO mode and cooling" by sending us a 0 and "I am in AUTO mode and heating" by sending a 7.

When we are after boot and in a steady state and it needs to cool according to the temperature, I can send 0,1 or 7, it always gives back the 0. When it needs to heat, it always gives back the 7.

My conclusion for the binding-logic is the following. At least for the ac_unit,
- (getter) sending a 1 is always the right behavior for writing.
- (setter) getting a 1 means, "I am in AUTO mode by cannot tell you what I am doing". Getting a 0 means "I am in AUTO mode and cooling" and getting a 7 means "I am in AUTO mode and heating"
- AUTO detection as build now is not good, as it makes 7 as value for the modern ones due to boot behavior. However, this causes heating for a couple of minutes even it is hot. This is somehow annoying. 

This leads two two questions:
- Maybe your airbase is following the same logic. Could you test that?
- Can you build the logic in the binding accordingly?

**Comment by jimtng** [comment]
Thanks for testing.

My airbase is completely different, that's why it uses a different thing id `airbase_ac_unit` instead of `ac_unit`, and also a different handler. For Airbase, AUTO=3, heating=1, cooling=2, Fan only=0. It kinda makes more sense in binary.

I can build the binding to the logic you described, and I'm re-stating it to my understanding here:

- When the adapter returns the status with mode=0|1|7 (from get_control_info), update the linked Mode item to AUTO
- Always send mode=1 to the adapter (with set_control_info) when we want to set the aircon to AUTO, regardless of whether the adapter reported mode=0, 1, or 7 in get_control_info

---

Now, I am a bit hesitant to change the binding's behaviour to send mode=1 for auto, in case it breaks other models currently in use. I would like to believe it won't, but a change is a change, and unless it's tested, it would be a problem for everyone. In the ideal world, I'd like to change it to mode=1 but I'd rather be conservative in this case, because I cannot test it.

So what I would propose is this:
- Remove the logic that remembers what get_control_info says (i.e. so when it says mode=7, don't use that as the auto mode number.
- By default, mode=0 is used for auto, but if we receive an error (`ret=PARAM NG`), then send mode=1 for auto, and remember this, so that in the future, it will send mode=1 instead of trying mode=0 first.

So this means that your aircon that supports mode=0 will continue to be sent mode=0 for auto - the exact same behaviour of the binding prior to the new PR, and your aircon that only supports mode=1 (and not 0 or 7) will receive mode=1.

How does this sound?

**Comment by LuiSauberhorn** [comment]
I think that makes total sense. I think when someone activates the aircon, it is usually about cooling, so your proposal should work fine in most cases anyway.

For mid-term, the interesting question would be do we find some testers who could try out manually the hypothesis. It is about one http request, should not be too complex. Maybe we put something into the forum, ask people to send one manual request with mode=1 and pow=1 when the aircon is out and ask 1) whether the aircon switches on and 2) how the reply looks like. What do you think?

**Comment by jimtng** [comment]
> For mid-term, the interesting question would be do we find some testers who could try out manually the hypothesis. It is about one http request, should not be too complex. Maybe we put something into the forum, ask people to send one manual request with mode=1 and pow=1 when the aircon is out and ask 1) whether the aircon switches on and 2) how the reply looks like. What do you think?

Sounds like a plan! Tag me (JimT) on that post! ;)

**Comment by jimtng** [comment]
@LuiSauberhorn please test v0.6 Version `4.1.0.202308201058`
https://github.com/jimtng/openhab-addons/releases/download/daikin-automode-0.6/org.openhab.binding.daikin-4.1.0-SNAPSHOT.jar

**Comment by LuiSauberhorn** [comment]
No does not work, I get:

```
2023-08-20 15:18:52.749 [WARN ] [org.apache.felix.fileinstall        ] - Error while starting bundle: file:/usr/share/openhab/addons/org.openhab.binding.daikin-4.1.0-SNAPSHOT.jar
org.osgi.framework.BundleException: Could not resolve module: org.openhab.binding.daikin [27]
  Unresolved requirement: Import-Package: javax.measure; version="[2.2.0,3.0.0)"

	at org.eclipse.osgi.container.Module.start(Module.java:463) ~[org.eclipse.osgi-3.18.0.jar:?]
	at org.eclipse.osgi.internal.framework.EquinoxBundle.start(EquinoxBundle.java:445) ~[org.eclipse.osgi-3.18.0.jar:?]
	at org.apache.felix.fileinstall.internal.DirectoryWatcher.startBundle(DirectoryWatcher.java:1260) ~[?:?]
	at org.apache.felix.fileinstall.internal.DirectoryWatcher.startBundles(DirectoryWatcher.java:1233) ~[?:?]
	at org.apache.felix.fileinstall.internal.DirectoryWatcher.startAllBundles(DirectoryWatcher.java:1221) ~[?:?]
	at org.apache.felix.fileinstall.internal.DirectoryWatcher.doProcess(DirectoryWatcher.java:515) ~[?:?]
	at org.apache.felix.fileinstall.internal.DirectoryWatcher.process(DirectoryWatcher.java:365) ~[?:?]
	at org.apache.felix.fileinstall.internal.DirectoryWatcher.run(DirectoryWatcher.java:316) ~[?:?]
```

**Comment by jimtng** [comment]
Ah, I'm afraid it might need 4.1-snapshot :( And.... right now the snapshot build is broken. I've been waiting for the next build, probably in 8 hours. Which openHAB version are you running right now?

**Comment by LuiSauberhorn** [comment]
openHAB 4.0.2

**Comment by jimtng** [comment]
@LuiSauberhorn please re-download https://github.com/jimtng/openhab-addons/releases/download/daikin-automode-0.6/org.openhab.binding.daikin-4.1.0-SNAPSHOT.jar

Version should say `4.1.0.202308201332`

**Comment by LuiSauberhorn** [comment]
Yes, bundle shows `org.openhab.binding.daikin:4.1.0.202308201332`

At the beginning after powering up, there is an unexpected behavior. But could be that the aircon controller was not completely online. 
```
2023-08-20 15:49:46.719 [TRACE] [ing.daikin.internal.DaikinWebTargets] - Calling url: http://192.168.191.128/aircon/set_control_info?mode=0&f_dir_ud=0&f_rate=A&f_dir_lr=0&shum=&pow=0&stemp=20.0
2023-08-20 15:49:46.729 [TRACE] [ing.daikin.internal.DaikinWebTargets] - Calling url: http://192.168.191.128/aircon/set_control_info?mode=1&f_dir_ud=0&f_rate=A&f_dir_lr=0&shum=&pow=0&stemp=20.0
2023-08-20 15:49:46.743 [WARN ] [internal.handler.DaikinAcUnitHandler] - AUTO mode not accepted with mode=0 or mode=1
```

A minute later, it looks good:
```
2023-08-20 15:50:49.005 [TRACE] [ing.daikin.internal.DaikinWebTargets] - Calling url: http://192.168.191.128/aircon/set_control_info?mode=0&f_dir_ud=0&f_rate=A&f_dir_lr=0&shum=&pow=0&stemp=23.0
2023-08-20 15:50:49.016 [TRACE] [ing.daikin.internal.DaikinWebTargets] - Calling url: http://192.168.191.128/aircon/set_control_info?mode=1&f_dir_ud=0&f_rate=A&f_dir_lr=0&shum=&pow=0&stemp=23.0
2023-08-20 15:50:49.035 [DEBUG] [internal.handler.DaikinAcUnitHandler] - AUTO uses mode=1
```

I can try it once again in a couple of hours but now need a moment to look after the kids...

**Comment by jimtng** [comment]
> But then it sets anyway mode=0

Thanks! You've found a bug!

Please try v0.7 Version `4.1.0.202308201432`
https://github.com/jimtng/openhab-addons/releases/download/daikin-automode-0.7/org.openhab.binding.daikin-4.1.0-SNAPSHOT.jar

Thanks for bearing with me through these tests!

**Comment by LuiSauberhorn** [comment]
No worries, never for the community, pure own interest ;-)

**Comment by LuiSauberhorn** [comment]
OK, tested it. I did not find a DEBUG / TRACE message that specifies how AUTO mode is selected but the result looks good. I will observe that over the next days and keep the trace running.
Next step is to start a thread to find some testers for the mode=1 path.
Again, thanks for the effort, much appreciated!

**Comment by LuiSauberhorn** [comment]
Post is out: https://community.openhab.org/t/daikin-binding-test-of-auto-mode-value-we-need-your-help/148919

**Comment by jimtng** [comment]
> I did not find a DEBUG / TRACE message that specifies how AUTO mode is selected

Thanks for testing! The debug log should be there, only once though, so maybe restart openhab or just stop / start the binding, then send an openhab command to change the mode to Auto and the log should be made.

**Comment by LuiSauberhorn** [comment]
Quite sure that the log statement for the AUTO mode is not there. Here you see the relevant log.

```
2023-08-22 20:34:53.124 [TRACE] [ing.daikin.internal.DaikinWebTargets] - Calling url: http://192.168.191.128/aircon/get_control_info
2023-08-22 20:34:53.139 [TRACE] [ding.daikin.internal.api.ControlInfo] - Parsing string: "ret=SERIAL IF FAILURE"
```
Here you see that the device is still not online...

```
2023-08-22 20:34:54.711 [TRACE] [n.internal.handler.DaikinBaseHandler] - Polling for state
2023-08-22 20:34:54.712 [TRACE] [ing.daikin.internal.DaikinWebTargets] - Calling url: http://192.168.191.129/aircon/get_control_info
2023-08-22 20:34:54.822 [TRACE] [ding.daikin.internal.api.ControlInfo] - Parsing string: "ret=OK,pow=0,mode=1,adv=,stemp=22.0,shum=0,dt1=22.0,dt2=M,dt3=25.0,dt4=25.0,dt5=25.0,dt7=22.0,dh1=0,dh2=50,dh3=0,dh4=0,dh5=0,dh7=0,dhh=50,b_mode=1,b_stemp=22.0,b_shum=0,alert=255,f_rate=A,f_dir=0,b_f_rate=A,b_f_dir=0,dfr1=A,dfr2=5,dfr3=5,dfr4=5,dfr5=5,dfr6=5,dfr7=A,dfrh=5,dfd1=0,dfd2=0,dfd3=0,dfd4=0,dfd5=0,dfd6=0,dfd7=0,dfdh=0,dmnd_run=0,en_demand=0"
2023-08-22 20:34:54.824 [TRACE] [ing.daikin.internal.DaikinWebTargets] - Calling url: http://192.168.191.129/aircon/get_sensor_info
2023-08-22 20:34:54.932 [TRACE] [nding.daikin.internal.api.SensorInfo] - Parsing string: "ret=OK,htemp=25.0,hhum=60,otemp=24.0,err=0,cmpfreq=0"
2023-08-22 20:34:54.937 [TRACE] [ing.daikin.internal.DaikinWebTargets] - Calling url: http://192.168.191.129/aircon/get_year_power_ex
2023-08-22 20:34:55.045 [TRACE] [g.daikin.internal.api.EnergyInfoYear] - Parsing string: "ret=OK,curr_year_heat=0/0/0/0/0/1/0/0/0/0/0/0,prev_year_heat=0/0/0/0/0/0/0/0/0/0/0/0,curr_year_cool=0/0/2/0/17/133/81/43/0/0/0/0,prev_year_cool=0/0/0/0/0/0/0/0/0/0/0/0"
2023-08-22 20:34:55.051 [TRACE] [ing.daikin.internal.DaikinWebTargets] - Calling url: http://192.168.191.129/aircon/get_week_power_ex
2023-08-22 20:34:55.161 [TRACE] [in.internal.api.EnergyInfoDayAndWeek] - Parsing string: "ret=OK,s_dayw=2,week_heat=0/0/0/0/0/0/0/0/0/0/0/0/0/0,week_cool=0/0/12/12/5/9/3/2/0/0/0/0/0/0"
2023-08-22 20:35:13.141 [TRACE] [n.internal.handler.DaikinBaseHandler] - Polling for state
2023-08-22 20:35:13.143 [TRACE] [ing.daikin.internal.DaikinWebTargets] - Calling url: http://192.168.191.128/aircon/get_control_info
2023-08-22 20:35:13.156 [TRACE] [ding.daikin.internal.api.ControlInfo] - Parsing string: "ret=OK,pow=0,mode=1,stemp=23.0,shum=--,dt1=23.0,dt2=--,dt3=23.0,dt4=23.0,alert=0,adv=,f_rate=A,dfr1=3,dfr2=-,dfr3=3,dfr4=3,dfr6=3"
2023-08-22 20:35:13.162 [TRACE] [ing.daikin.internal.DaikinWebTargets] - Calling url: http://192.168.191.128/aircon/get_sensor_info
2023-08-22 20:35:13.176 [TRACE] [nding.daikin.internal.api.SensorInfo] - Parsing string: "ret=OK,htemp=25,otemp=-,hhum=-,err=0"
2023-08-22 20:35:13.181 [TRACE] [ing.daikin.internal.DaikinWebTargets] - Calling url: http://192.168.191.128/aircon/get_year_power_ex
2023-08-22 20:35:13.191 [DEBUG] [ing.daikin.internal.DaikinWebTargets] - Daikin controller HTTP status: 404 - Not Found http://192.168.191.128/aircon/get_year_power_ex
2023-08-22 20:35:13.192 [TRACE] [g.daikin.internal.api.EnergyInfoYear] - Parsing string: "ret=PARAM NG,msg=404 Not Found"
2023-08-22 20:35:13.194 [TRACE] [ing.daikin.internal.DaikinWebTargets] - Calling url: http://192.168.191.128/aircon/get_week_power_ex
2023-08-22 20:35:13.203 [DEBUG] [ing.daikin.internal.DaikinWebTargets] - Daikin controller HTTP status: 404 - Not Found http://192.168.191.128/aircon/get_week_power_ex
2023-08-22 20:35:13.204 [TRACE] [in.internal.api.EnergyInfoDayAndWeek] - Parsing string: "ret=PARAM NG,msg=404 Not Found"
2023-08-22 20:35:13.206 [DEBUG] [in.internal.api.EnergyInfoDayAndWeek] - EnergyInfoDayAndWeek::parse() did not receive 'ret=OK' from adapter
```
Until here, .128 is not online. Here, we get the first reply:

```
2023-08-22 20:35:13.235 [TRACE] [ing.daikin.internal.DaikinWebTargets] - Calling url: http://192.168.191.128/aircon/get_control_info
2023-08-22 20:35:13.246 [TRACE] [ding.daikin.internal.api.ControlInfo] - Parsing string: "ret=OK,pow=0,mode=1,stemp=23.0,shum=--,dt1=23.0,dt2=-
```
Here, we may know that the mode=1 is the right one, but no log entry

```
-,dt3=23.0,dt4=23.0,alert=0,adv=,f_rate=A,dfr1=3,dfr2=-,dfr3=3,dfr4=3,dfr6=3"
2023-08-22 20:35:13.252 [TRACE] [ing.daikin.internal.DaikinWebTargets] - Calling url: http://192.168.191.128/aircon/set_control_info?mode=1&f_dir_ud=0&f_rate=A&f_dir_lr=0&shum=&pow=1&stemp=23.0
2023-08-22 20:35:15.163 [TRACE] [n.internal.handler.DaikinBaseHandler] - Polling for state
```
However, the binding still uses the right value. So functional-wise, all works correctly.

Best, Lui
